src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -186,6 +207,93 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree,
 
 
 /*
+ * ReplaceExternParamsInOriginalQuery replaces the external parameters that appears
+ * in the query with the corresponding entries in the boundParams.
+ *
+ * Note that this function is inspired by eval_const_expr() on Postgres. We cannot
+ * use that function because it requires access to PlannerInfo.
+ */
+static Node *
+ReplaceExternParamsInOriginalQuery(Node *inputNode, ParamListInfo boundParams)
+{
+	int parameterIndex = 0;
+
+	Assert(boundParams != NULL);
+
+	if (inputNode == NULL)
+	{
+		return NULL;
+	}
+
+	if (IsA(inputNode, Param))
+	{
+		Param *paramToProcess = (Param *) inputNode;
+		ParamExternData correspondingParameterData;
```

Comment:
Ok, that probably makes more sense. That is also how it is used in `eval_const_expressions_mutator()`.